ontotext-yasgui-web-component/src/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,47 @@
+import {Component, Event, EventEmitter, h, Prop, State} from '@stencil/core';
+import {TranslationService} from '../../services/translation.service';
+import {DropdownOption} from '../../models/dropdown-option';
+import {InternalDropdownValueSelectedEvent} from '../../models/internal-events/internal-dropdown-value-selected-event';
+
+@Component({
+  tag: 'ontotext-yasgui-dropdown',
+  styleUrl: 'dropdown.scss',
+  shadow: false,
+})
+export class Dropdown {
+
+  @State() open = false;
+
+  @Prop() translationService: TranslationService;
+  @Prop() nameLabelKey: string;
+  @Prop() items: DropdownOption[];
+
+  @Event() valueChanged: EventEmitter<InternalDropdownValueSelectedEvent>;
+
+  private onSelect(value: string) {
+    this.open = false;
+    this.valueChanged.emit(new InternalDropdownValueSelectedEvent(value));
+  }
+
+  private toggleComponent(): void {
+    this.open = !this.open;
+  }
+
+  render() {
+    return (
+      <div class="ontotext-yasgui-dropdown">
+        <button class={`ontotext-yasgui-dropdown-button ${this.open ? 'icon-caret-up-after' : ' icon-caret-down-after'}`}
+                onClick={() => this.toggleComponent()}>
+          {this.translationService.translate(this.nameLabelKey)}
+        </button>
+
+        <ul class={`ontotext-yasgui-dropdown-menu ${this.open ? 'open' : 'closed'}`}>
```

Comment:
This is not quite responsive. If there is not enough space on the right a horizontal scrollbar appears. It's probably not important to be handled right now, I just wanted to mention it.